### PR TITLE
Remove reusable connections term from charter

### PR DESCRIPTION
**Pull request by relu91**
If you want to keep a detailed list of possible features implemented as an improvement of the descriptiveness of TDs, I think using `stateful interactions` captures better the summary of the different discussions that we had. I am talking about stateful interactions because currently, we treat each operation as stateless meaning that the Consumer can invoke one operation without performing any initial setup or keeping track of how many times he called that ops. Note this is also true for subscribe ops (exceptions could be unsubscribe ops and the recently added async actions but we treated them similarly to the other ops causing issues). 

With stateful interactions, we can capture protocols that need to keep track of the state of the communication channel between the two parties (like persistent connections or even complex rendez-vous). On the other hand, it is a more ambitious objective to achieve.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-charter-drafts/pull/73.html" title="Last updated on Mar 1, 2023, 5:25 PM UTC (19c35fe)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-charter-drafts/73/04cdada...19c35fe.html" title="Last updated on Mar 1, 2023, 5:25 PM UTC (19c35fe)">Diff</a>

### CODE REVIEW

**Comment by benfrancis** [comment]
The specific use case I personally have in mind for this is that multiple interaction affordances can share the same WebSocket connection. Ideally that WebSocket endpoint only needs defining once in the Thing Description and can be re-used by multiple interaction affordances, but the important point is that a Consumer does not need to open a new WebSocket connection for every interaction.

See https://github.com/w3c/wot-thing-description/issues/1070, https://github.com/w3c/wot-thing-description/issues/878 and https://github.com/w3c/wot-thing-description/issues/1664.

I actually think the term "reusable connections" captures that better than "stateful interactions" since the interaction itself is not "stateful" in that it doesn't necessarily care about other interactions, it's just that the connection is re-used between interactions.

It sounds like the idea of "stateful interactions" might describe other use cases, but I'm not sure what they are.

**Comment by mmccool** [comment]
I think these are talking about two different things.  Should be discussed in the TD call.  Note one option is not mentioning it in the charter summary, which might be for the best rather than having confusing underdefined terminology.

**Comment by relu91** [comment]
> It sounds like the idea of "stateful interactions" might describe other use cases, but I'm not sure what they are.

It is by purpose a more general concept that in my mind includes also the use cases that you mentioned. I'll try to collect relevant issues here, but I recall discussing possible similarities with some sec protocols that require a pre-amble interaction before using resources or even async actions that require a step-by-step flow of primitive calls to obtain the desired result.

> I actually think the term "reusable connections" captures that better than "stateful interactions" since the interaction itself is not "stateful" in that it doesn't necessarily care about other interactions, it's just that the connection is re-used between interactions.

I think the fact that you are **reusing** a connection proves already the fact that a Consumer implementer should keep track of the state of the communication (maybe communication is a better word here?) with the Remote Web Thing. 

Anyhow, this term might be too ambitions or not well received by the group and I am open to hearing more points on this.

**Comment by egekorkan** [comment]
TD Call of 01.03:
- @mjkoster : reusable connection for persistent endpoints. Connection context is another proposal.
- @relu91 : you are actually managing a connection state and it will help with scripting architecture as well
- @benfrancis : stateful interactions for async actions go in the same wording so might be confusing
- @ashimura and @mmccool : it would be fine to remove it since we say "includes" anyways.

We will remove the term from the charter, details document has information on it.

The two use cases are:
- Endpoint "copy pasting", i.e. the base URI we have at the moment
- Keeping a connection alive 

Alternative:
- reusable endpoint
- stateful connection
- persistent endpoints

File: wot-wg-2023-draft.html [hunk 1]
```diff
@@ -230,7 +230,7 @@ <h2>Scope</h2>
           </dt>
           <dd>
             The WG will work on increasing the descriptive power of TDs to allow using them in more use cases. This includes
-            describing dynamic resources, reusable connections, historical values (timeseries), repeating payloads, both static and dynamic
+            describing dynamic resources, stateful interactions, historical values (timeseries), repeating payloads, both static and dynamic
```
Comment (egekorkan) [inline wot-wg-2023-draft.html:233]:
```suggestion
            describing dynamic resources, historical values (timeseries), repeating payloads, both static and dynamic
```

**Comment by egekorkan** [comment]
@relu91 I have removed the term and changed the PR title

**Comment by benfrancis** [comment]
As I mentioned on the TD call, I would prefer to keep this but it's not a huge deal. We clearly have consensus that this is something we need and it's described in detail in the details document. I actually do think we have a common understanding of what "reusable connections" means and @relu91 was just trying to encompass more use cases with a broader term.

Anyway, hopefully a final decision will be made on the call tomorrow but I don't strongly object to removing it if someone feels strongly that it needs to be removed.

**Pull request merged**